src/android/io/sqlc/SQLitePlugin.java

```diff
@@ -256,7 +264,15 @@ private SQLiteAndroidDatabase openDatabase(String dbname, String key, CallbackCo
             return mydb;
         } catch (Exception e) {
             // NOTE: NO Android locking/closing BUG workaround needed here
-            cbc.error("can't open database " + e);
+
+            if(selfHealingEnabled && e.getMessage().contains("file is encrypted or is not a database:")) {
+                logger.logError("Android database will be deleted to self heal:" + e.getMessage(), "SQLite");
+                deleteDatabaseNow(dbname);
+                return openDatabase(dbname, key, cbc, false);
+            } else {
+                cbc.error("can't open database " + e);
```

Comment:
Since we are adding the OS Logger in the plugin, don't we want to log this error message too?

Reply:
@vmfo22 I agree with Luis, now that we use the logger plugin here, we should also see what other stuff we should also log. It makes sense to open a new issue for this?